### PR TITLE
Caleb waters

### PR DESCRIPTION
**Pull request by Mekanation**
I did a commit for each test so you could see the whole process.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>RELEASE</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
```
Comment (rbussiere-tek) [inline pom.xml:34]: Not needed and ambiguous.

File: src/main/java/Carts.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import java.util.ArrayList;
+
+public class Carts {
```
Comment (rbussiere-tek) [inline src/main/java/Carts.java:3]: Should be singular.

File: src/main/java/Products.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+public class Products {
```
Comment (rbussiere-tek) [inline src/main/java/Products.java:1]: Should be singular.

File: src/main/java/Products.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    private Products(int productID, String productName, double productPrice, boolean exempt, boolean imported) {
+        this.productID = productID;
+        this.productName = productName;
+        this.productPrice = productPrice;
+        this.exempt = exempt;
+        this.imported = imported;
+    }
+
+    public static final Products book = new Products(1,"Book",12.49, true, false);
```
Comment (rbussiere-tek) [inline src/main/java/Products.java:17]: static final could be all UPPERCASE. (I know there are multiple schools of thought on this but I must be old fashioned... what can I do???

File: src/main/java/Carts.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    private int ID;
```
Comment (rbussiere-tek) [inline src/main/java/Carts.java:4]: Should be all lowercase.

File: src/main/java/Carts.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    private String name;
+    private ArrayList<Products> productsInCart;
+    private double total = 0.00;
```
Comment (rbussiere-tek) [inline src/main/java/Carts.java:7]: Should be a local variable instead.

File: src/main/java/Carts.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    private double taxes = 0.00;
```
Comment (rbussiere-tek) [inline src/main/java/Carts.java:8]: Should be a local variable instead.

File: src/main/java/Products.java (hunk 3 above)
Comment (rbussiere-tek) [inline src/main/java/Products.java:1]: Look at proposed changes in attachment, tests could have to be adjusted to account for corrected totals
[Products.txt](https://github.com/Mekanation/tdd/files/4955267/Products.txt)

**Review by rbussiere-tek — APPROVED** [review]
- Runs fine? (Wrong results on output -> see suggestions)
- Code easy to read and to maintain (i.e. Clean code)? +4
- Object-oriented design? +5
- Solid Principles? +4
- TDD? +3 (tests centralized in 1 class, every business class deserves its own UT, ordering tests is bad practice)
- Expandable design? +4

File: src/main/java/Products.java (hunk 4 above)
Comment (Mekanation) [reply src/main/java/Products.java:17]: This is how intellij autofills.

Comment (rbussiere-tek) [reply src/main/java/Products.java:17]: Ahhh never rely on an IDE lol!

**Pull request merged**